blockly/src/entities/BlocklyMonster.java

```diff
@@ -0,0 +1,177 @@
+      0,
+      MonsterIdleSound.BURP);
+
+  private final String name;
+  private final IPath texture;
+  private final Sound deathSound;
+  private final Supplier<Consumer<Entity>> fightAISupplier;
+  private final Supplier<Consumer<Entity>> idleAISupplier;
+  private final Supplier<Function<Entity, Boolean>> transitionAISupplier;
+  private final int collideDamage;
+  private final int collideCooldown;
+  private final IPath idleSoundPath;
+  private final int health;
+  private final float speed;
+  private final float itemChance; // 0.0f means no items, 1.0f means always items
+
+  /**
+   * Creates a new MonsterType with the given parameters.
+   *
+   * @param name The name of the monster.
+   * @param texture The path to the texture to use for the monster.
+   * @param health The amount of health the monster has.
+   * @param speed The speed of the monster.
+   * @param canHaveItems The chance that the monster will drop an item upon death. If 0, no item
+   *     will be dropped. If 1, an item will always be dropped.
+   * @param deathSound The sound to play when the monster dies.
+   * @param fightAISupplier The supplier for the fight AI.
+   * @param idleAISupplier The supplier for the idle AI.
+   * @param transitionAISupplier The supplier for the transition AI.
+   * @param collideDamage The damage the monster inflicts upon collision.
+   * @param collideCooldown The cooldown time between monster's collision damage.
+   * @param idleSound The sound to play when the monster is idle.
+   */
+  BlocklyMonster(
+      String name,
+      String texture,
+      int health,
+      float speed,
+      float canHaveItems,
+      MonsterDeathSound deathSound,
+      Supplier<Consumer<Entity>> fightAISupplier,
+      Supplier<Consumer<Entity>> idleAISupplier,
+      Supplier<Function<Entity, Boolean>> transitionAISupplier,
+      int collideDamage,
+      int collideCooldown,
+      MonsterIdleSound idleSound) {
+    this.name = name;
+    this.texture = new SimpleIPath(texture);
+    this.health = health;
+    this.speed = speed;
+    this.itemChance = canHaveItems;
+    this.deathSound = deathSound.sound();
+    this.fightAISupplier = fightAISupplier;
+    this.idleAISupplier = idleAISupplier;
+    this.transitionAISupplier = transitionAISupplier;
+    this.collideDamage = collideDamage;
+    this.collideCooldown = collideCooldown;
+    this.idleSoundPath = idleSound.path();
+  }
+
+  /**
+   * Builds a monster entity with the given parameters.
+   *
+   * @param spawnPos The position where the monster should spawn.
+   * @param viewDirection The direction the monster should face.
+   * @return A new Entity representing the monster.
+   * @throws IOException if the animation could not be loaded.
+   * @see MonsterFactory#buildMonster(String, IPath, int, float, float, Sound, AIComponent, int,
+   *     int, IPath) MonsterFactory.buildMonster
+   */
+  public Entity buildMonster(Point spawnPos, PositionComponent.Direction viewDirection)
```

Comment:
kann ich mir hier noch nen `int range` Parameter wünschen? 
Das wird in Blockly tatsächlich wichtig.